services/workflows-service/src/alert/alert.service.ts

```diff
@@ -1,25 +1,35 @@
+import { map } from 'rxjs/operators';
 import { AlertRepository } from '@/alert/alert.repository';
 import { AppLoggerService } from '@/common/app-logger/app-logger.service';
 import * as errors from '@/errors';
 import { PrismaService } from '@/prisma/prisma.service';
 import { isFkConstraintError } from '@/prisma/prisma.util';
 import { ObjectValues, TProjectId } from '@/types';
 import { Injectable } from '@nestjs/common';
-import { Alert, AlertDefinition, AlertState, AlertStatus } from '@prisma/client';
+import { Alert, AlertDefinition, AlertState, AlertStatus, Prisma } from '@prisma/client';
 import { CreateAlertDefinitionDto } from './dtos/create-alert-definition.dto';
 import { FindAlertsDto } from './dtos/get-alerts.dto';
+import { DataAnalyticsService } from '@/data-analytics/data-analytics.service';
+import { AlertDefinitionRepository } from '@/alert-definition/alert-definition.repository';
+import { InlineRule } from '@/data-analytics/types';
+import _ from 'lodash';
+
+// TODO: move to utils
 
 @Injectable()
 export class AlertService {
   constructor(
     private readonly prisma: PrismaService,
     private readonly logger: AppLoggerService,
+    private readonly dataAnalyticsService: DataAnalyticsService,
     private readonly alertRepository: AlertRepository,
+    private readonly alertDefinitionRepository: AlertDefinitionRepository,
   ) {}
 
   async create(dto: CreateAlertDefinitionDto, projectId: TProjectId): Promise<AlertDefinition> {
+    const _data = { ...dto, projectId };
```

Comment:
```suggestion
    const dataWithoutProjectId = { ...dto, projectId: _projectId };
```

Reply:
i dont want to exclude proejectId,
appended projectId with dto properties